subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -3215,6 +3215,10 @@ static int tls_sock_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 		return ztls_poll_offload(fds, nfds, timeout);
 	}
 
+	case ZFD_IOCTL_FIONBIO:
+	case ZFD_IOCTL_FIONREAD:
+		return zsock_ioctl(ctx->sock, request, args);
```

Comment:
I don't think that'd be correct for `ZFD_IOCTL_FIONREAD` on TLS socket. The underlying TCP data length may no equal to data length after decryption. We have a dedicated function, `ztls_socket_data_check()`, to check data available after processing by mbed TLS.

Reply:
Definitely - need to add some test coverage too. I'll see if I can get to this a bit later today.

Reply:
@rlubos - I think I'll drop the TLS changes for now. `ztls_socket_data_check()` consistently returns `-EIO` due to `mbedtls_ssl_read()` returning `MBEDTLS_ERR_SSL_BAD_INPUT_DATA` in the case of DTLS.

After looking around, I realized that TLS sockets do not support "peek" operations and would require an intermediate buffer of plain text data in order to determine the size, which is maybe not ideal.

I think that's the case for both TLS and DTLS.

I don't have time right now to look into a possible workaround, but I'll attach the test changes I was planning on adding for TLS to this PR.

Reply:
I don't see anything obvious why would DTLS report an error on `ztls_socket_data_check()`, this needs investigating. I'm generally planning to extend TLS/DTLS tests to cover more socket functionalities, I'm just not there yet. 

In any case, I'm ok with skipping TLS/DTLS sockets implementation as for now.